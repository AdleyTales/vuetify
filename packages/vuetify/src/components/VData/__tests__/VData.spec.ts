import VData from '../VData'
import {
  mount,
  MountOptions,
  Wrapper
} from '@vue/test-utils'

describe('VData.ts', () => {
  type Instance = InstanceType<typeof VData>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(VData, options)
    }
  })

  it('should render with scopedProps', async () => {
    const wrapper = mountFunction({
      computed: {
        scopedProps: () => 'test'
      },
      scopedSlots: {
        default (data) {
          return this.$createElement('div', [ data ])
        }
      }
    })

    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should compute length', () => {
    const wrapper = mountFunction({
      propsData: {
        items: [ 'foo', 'bar' ]
      }
    })

    expect(wrapper.vm.itemsLength).toBe(2)

    wrapper.setProps({
      serverItemsLength: 10
    })

    expect(wrapper.vm.itemsLength).toBe(10)
  })
})
