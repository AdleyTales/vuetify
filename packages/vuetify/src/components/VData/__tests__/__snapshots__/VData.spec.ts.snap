// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VData.ts should render with scopedProps 1`] = `

<div>
  test
</div>

`;
